miniapps/shifted_methods/dist_solver.hpp

```diff
@@ -0,0 +1,322 @@
+//The negative part - negative unit volumetric loading
+//The parameter rh is the radius of a linear cone filter which will deliver
+//similar smoothing effect as the Screened Poisson euation
+//It determines the length scale of the smoothing.
+class ScreenedPoisson: public NonlinearFormIntegrator
+{
+protected:
+    double diffcoef;
+    mfem::Coefficient* func;
+
+public:
+    ScreenedPoisson(mfem::Coefficient& nfunc, double rh):func(&nfunc)
+    {
+        double rd=rh/(2*std::sqrt(3.0));
+        diffcoef= rd*rd;
+    }
+
+    ~ScreenedPoisson() { }
+
+    void SetInput(mfem::Coefficient& nfunc) { func = &nfunc; }
+
+    virtual double GetElementEnergy(const FiniteElement &el,
+                                    ElementTransformation &trans,
+                                    const Vector &elfun) override;
+
+    virtual void AssembleElementVector(const FiniteElement &el,
+                                       ElementTransformation &trans,
+                                       const Vector &elfun,
+                                       Vector &elvect) override;
+
+    virtual void AssembleElementGrad(const FiniteElement &el,
+                                        ElementTransformation &trans,
+                                        const Vector &elfun,
+                                        DenseMatrix &elmat) override;
+};
+
+
+///the vector coefficent should return a vector eith elements:
+/// [0] - derivative with respect to x
+/// [1] - derivative with respect to y
+/// [2] - derivative with respect to z
+class PUMPLaplacian: public NonlinearFormIntegrator
+{
+
+protected:
+    mfem::Coefficient *func;
+    mfem::VectorCoefficient *fgrad;
+    bool ownership;
+    double pp, ee;
+
+public:
+    PUMPLaplacian(Coefficient* nfunc, VectorCoefficient* nfgrad, bool ownership_=true)
+    {
+        func=nfunc;
+        fgrad=nfgrad;
+        ownership=ownership_;
+        pp=2.0;
+        ee=1e-7;
+    }
+
+    void SetPower(double pp_) { pp = pp_; }
+    void SetReg(double ee_)   { ee = ee_; }
+
+    virtual ~PUMPLaplacian()
+    {
+        if(ownership)
+        {
+            delete func;
+            delete fgrad;
+        }
+    }
+
+    virtual double GetElementEnergy(const FiniteElement &el,
+                                    ElementTransformation &trans,
+                                    const Vector &elfun) override;
+
+    virtual void AssembleElementVector(const FiniteElement &el,
+                                       ElementTransformation &trans,
+                                       const Vector &elfun,
+                                       Vector &elvect) override;
+
+    virtual void AssembleElementGrad(const FiniteElement &el,
+                                        ElementTransformation &trans,
+                                        const Vector &elfun,
+                                        DenseMatrix &elmat) override;
+
+};
+
+class PDEFilter
```

Comment:
`PDEFilter` is an intriguing name for a class.  A short comment block would be again be appreciated.